pkg/rtc/room.go

```diff
@@ -635,10 +635,9 @@ func (r *Room) subscribeToExistingTracks(p types.Participant) {
 
 // broadcast an update about participant p
 func (r *Room) broadcastParticipantState(p types.Participant, skipSource bool) {
-	r.lock.Lock()
```

Comment:
This lock looked redundant. So, removed it.

Reply:
this was added to prevent two goroutines from acquiring the exact timestamp/updates. If multiple requests come in to broadcastParticipantState, we would end up having a race that end up "erasing" one of the published tracks.

Since Go generates monotonic timestamps, the idea is to ensure the order of the content matches the ordering of Time. i.e. if a timestamp is later, then the accompanying `updates` is also more recent.

there may be better ways of preventing this race condition.

Reply:
Ouch, I thought it was locking some state and was forgotten when some code was removed. Will add it back and add a comment around it too.